vllm/utils.py

```diff
@@ -42,3 +43,49 @@ def random_uuid() -> str:
 def in_wsl() -> bool:
     # Reference: https://github.com/microsoft/WSL/issues/4071
     return "microsoft" in " ".join(uname()).lower()
+
+
+def get_max_shared_mem_bytes() -> int:
+    """Gets the max optin shared memory per block for the current GPU."""
+    libnames = ("libcuda.so", "libcuda.dylib", "nvcuda.dll", "cuda.dll")
+    for libname in libnames:
+        try:
+            cuda = ctypes.CDLL(libname)
+        except OSError:
+            continue
+        else:
+            break
+    else:
+        raise OSError(f"Could not load any of {' '.join(libnames)}")
```

Comment:
I'm not sure how robust this is. Why don't we make a pytorch custom op (like `cuda_ops`) that executes the logic below in C++/CUDA?

Reply:
Sure, wanted to move fast on this but a custom op would work too